Cython/Compiler/PyrexTypes.py

```diff
@@ -1504,16 +1504,20 @@ def type_check_function(self, exact=True):
     def isinstance_code(self, arg):
         return '%s(%s)' % (self.type_check_function(exact=False), arg)
 
-    def type_test_code(self, arg, notnone=False, exact=True):
+    def type_test_code(self, scope, arg, notnone=False, exact=True):
+        # returns code_str, utility_code
         type_check = self.type_check_function(exact=exact)
         check = 'likely(%s(%s))' % (type_check, arg)
+        utility_code = UtilityCode.load_cached(
+                    "RaiseUnexpectedTypeError", "ObjectHandling.c")
         if not notnone:
             check += '||((%s) == Py_None)' % arg
         if self.name == 'basestring':
             name = '(PY_MAJOR_VERSION < 3 ? "basestring" : "str")'
         else:
             name = '"%s"' % self.name
-        return check + ' || __Pyx_RaiseUnexpectedTypeError(%s, %s)' % (name, arg)
+        return (check + ' || __Pyx_RaiseUnexpectedTypeError(%s, %s)' % (name, arg),
+                utility_code)
```

Comment:
Returning the utility code should fix https://github.com/cython/cython/issues/6320

Reply:
It looks like this is called too late to call `env.use_utility_code`

Reply:
I don't think it's too late. The tests that fail due to this seem to be mostly fused types and memory view tests, i.e. those for which we generate Cython utility code. My guess is that end up with a the Cython utility scope here and that the utility code that it uses doesn't get passed on to the module scope.